### PR TITLE
DM-45719: Add github action to build arm64 as well as x86

### PR DESCRIPTION
**Pull request by roceb**
- Added a GitHub Action to automate CI/CD workflows
- Added Docker support for ARM64 architecture
- Removed unused test files and old Travis CI configuration files
- Fixed YAML linting issues 
- Updated the README to reflect the new changes

### CODE REVIEW

File: .github/workflows/yamllint.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+---
```
Comment (ktlim) [inline .github/workflows/yamllint.yaml:1]: We can't use https://github.com/lsst/rubin_workflows/blob/main/.github/workflows/yamllint.yaml ?  Should be similar to the rebase_checker.

File: .github/workflows/shellcheck.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+---
```
Comment (ktlim) [inline .github/workflows/shellcheck.yaml]: Is there any shell left to check now that you've gotten rid of the test shell scripts?

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push
+        uses: docker/build-push-action@v6
+        with:
+          platforms: linux/amd64,linux/arm64
```
Comment (ktlim) [inline .github/workflows/build.yaml:21]: We might see if we can get SQuaRE to add a `platforms` argument to their https://github.com/lsst-sqre/build-and-push-to-ghcr action; that might simplify this.

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+          push: true
+          tags: ghcr.io/${{ github.repository }}
```
Comment (ktlim) [inline .github/workflows/build.yaml]: The [pipeline that this is replacing](https://github.com/lsst-dm/jenkins-dm-jobs/blob/main/pipelines/sqre/infra/build_docker_gc.groovy#L45-L70) adds a `latest` tag and defeats the build cache.  Not sure if we need those here.

**Review by ktlim — APPROVED** [review]

File: .github/workflows/build.yaml (hunk 3 above)
Comment (roceb) [reply .github/workflows/build.yaml:21]: Created [build-and-push](https://github.com/lsst-dm/build-and-push) to add a few other features that we use like label, platform and build-args.

**Pull request merged**